slasher/rpc/server.go

```diff
@@ -17,17 +25,27 @@ import (
 // Server defines a server implementation of the gRPC Slasher service,
 // providing RPC endpoints for retrieving slashing proofs for malicious validators.
 type Server struct {
-	ctx       context.Context
-	detector  *detection.Service
-	slasherDB db.Database
+	ctx          context.Context
+	detector     *detection.Service
+	slasherDB    db.Database
+	beaconClient *beaconclient.Service
 }
 
 // IsSlashableAttestation returns an attester slashing if the attestation submitted
 // is a slashable vote.
 func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.IndexedAttestation) (*slashpb.AttesterSlashingResponse, error) {
 	ctx, span := trace.StartSpan(ctx, "detection.IsSlashableAttestation")
 	defer span.End()
-	//TODO(#5189) add signature validation to prevent DOS attack on the endpoint.
+	valid, err := ss.verifyIndexedAttestationSig(ctx, req)
+	if err != nil {
+		log.WithError(err).Error("Failed to verify indexed attestation signature")
+		return nil, status.Errorf(codes.Internal, "Could not verify indexed attestation signature: %v: %v", req, err)
+	}
+	if !valid {
+		log.Warning("Indexed attestation signature failed to verify")
+		return nil, status.Errorf(codes.Unauthenticated, "Indexed attestation signature failed to verify: %v", req)
```

Comment:
```suggestion
		return nil, status.Errorf(codes.Internal, "Indexed attestation signature failed to verify: %v", req)
```

Reply:
Unauthenticated is typically used for user auth requests such as not having a proper cookie or auth header, etc.

Reply:
thanks, changed